### PR TITLE
DAOS-9679 mercury: Add UCX support to DAOS

### PR DESCRIPTION
**Pull request by jgmoore-or**
Updates DAOS config files to pull the latest Mercury RPM (support UCX) and to utilize the patch to the UCX network adapter code to:

- Modify parsing of address string
- Support listener registration in dual-port deployments.
- Change to active messaging API for two-side communication.

Signed-off-by: Joseph Moore <joseph.moore@intel.com>

### CODE REVIEW

File: utils/rpms/daos.spec [hunk 1]
```diff
@@ -539,6 +539,9 @@ getent passwd daos_agent >/dev/null || useradd -s /sbin/nologin -r -g daos_agent
 # No files in a shim package
 
 %changelog
+* Sat Apr  2 03:56:35 UTC 2022 Joseph Moore <joseph.moore@intel.com> 2.1.100-25
```
Comment (brianjmurrell) [inline utils/rpms/daos.spec:542]: You need to update the `Release:` to reflect the bump to `25`.  You also need to add an entry to [debian/changelog](https://github.com/daos-stack/daos/blob/433f13fb8c6a6fa420a74324e02257f1bbd0d3e8/debian/changelog) and update the minimum mercury version for the DEB build: https://github.com/daos-stack/daos/blob/433f13fb8c6a6fa420a74324e02257f1bbd0d3e8/debian/control#L13

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

File: utils/rpms/daos.spec [hunk 2]
```diff
@@ -3,7 +3,7 @@
 %define agent_svc_name daos_agent.service
 %define sysctl_script_name 10-daos_server.conf
 
-%global mercury_version 2.1.0~rc4-6%{?dist}
+%global mercury_version 2.1.0~rc4-7%{?dist}
```
Comment (brianjmurrell) [inline utils/rpms/daos.spec:6]: The latest is 8 now, not 7.

```suggestion
%global mercury_version 2.1.0~rc4-8%{?dist}
```

Comment (jgmoore-or) [reply utils/rpms/daos.spec]: All review comments have been addressed in the latest submission.

File: utils/rpms/daos.spec [hunk 3]
```diff
@@ -539,6 +539,9 @@ getent passwd daos_agent >/dev/null || useradd -s /sbin/nologin -r -g daos_agent
 # No files in a shim package
 
 %changelog
+* Sat Apr  2 2022 Joseph Moore <joseph.moore@intel.com> 2.1.100-25
+- Update to mercury 2.0.1.rc4-8 to include UCX provider patch
```
Comment (frostedcmos) [inline utils/rpms/daos.spec:543]: looks like we have typo here and in previous comments too. should be mercury 2.1.0rc4-8 not 2.0.1rc4-8

Comment (johannlombardi) [reply utils/rpms/daos.spec:543]: Please address

**Review by johannlombardi — CHANGES_REQUESTED** [review]

**Review by johannlombardi — DISMISSED** [review]

**Comment by johannlombardi** [comment]
Please inspect.

File: utils/rpms/daos.spec (hunk 1 above)
Comment (brianjmurrell) [reply utils/rpms/daos.spec]: The `Release:` is still at `24`:
https://github.com/daos-stack/daos/blob/7e22ccd58f940b9648d5b82fb86cf7221cd129c2/utils/rpms/daos.spec#L18

That needs to be `25`.  Unfortunately GitHub doesn't let me make code change suggestions for unmodified portions of files.

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

**Review by frostedcmos — APPROVED** [review]

**Review by brianjmurrell — APPROVED** [review]

**Pull request merged**